### ISSUE TITLE
Selecting a rowField as an array throws a NullPointerException

### ISSUE THREAD

**Issue by mjburghoffer** [jOOQ/jOOQ#7831]
### Expected behavior and actual behavior:
If I query a row as a field, the `DefaultBinding` throws a `NullPointerException`.  I would expect it to return results instead.

### Steps to reproduce the problem (include example code if possible):

This does not have all the supporting classes, etc, but is a snippet of code that is throwing the exception.

```kotlin
        val row = DSL.row(
            fieldsTable.fieldName,
            fieldsTable.fullType
        )

        val fieldsAgg = DSL.select(DSL.arrayAgg(DSL.rowField(row)))
            .from(fieldsTable)
            .where(fieldsFilter)
            .asField<Array<Record2<String, String>>>()

        val results = DSL.select(
            recordTable.organizationId,
            recordTable.id,
            recordTable.displayName,
            recordTable.description,
            recordTable.createdTime,
            recordTable.modifiedTime,
            recordTable.deletedTime,
            recordTable.objectSchemaNamespace,
            recordTable.objectSchemaName,
            fieldsAgg
        )
            .from(recordTable)
            .where(recordTable.id.eq(filter.id))
            .apply { attach(readerConfig) } // attach the data source config
            .fetchOne()
```

### Cause:
In `org.jooq.impl.DefaultBinding.pgNewRecord(...)`:

```java
            if (fields == null && type == Record.class)
                fields = Tools.fields(values.size(), SQLDataType.VARCHAR);
```

`type` is never `Record.class`, it will always be a concrete implementation of `Record` (often `RecordImpl`).  Perhaps this code would solve this problem:

```java
            if (fields == null && Record.class.isAssignableFrom(type))
                fields = Tools.fields(values.size(), SQLDataType.VARCHAR);
```

### Stack Trace:

```
09:12:57.918 [main @coroutine#1] ERROR org.jooq.impl.DefaultBinding - Cannot parse array       : {"(size,Integer)","(name,Text)","(value,Decimal)","(joined,DateTime)","(aliases,List<Text>)"}
java.lang.IllegalStateException: Could not construct new record
	at org.jooq.impl.Tools.newRecord(Tools.java:765)
	at org.jooq.impl.Tools.newRecord(Tools.java:748)
	at org.jooq.impl.Tools.newRecord(Tools.java:712)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.pgNewRecord(DefaultBinding.java:3017)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.get0(DefaultBinding.java:2857)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.get0(DefaultBinding.java:2778)
	at org.jooq.impl.DefaultBinding$AbstractBinding.get(DefaultBinding.java:774)
	at org.jooq.impl.DefaultBinding$DefaultArrayBinding.pgGetArray(DefaultBinding.java:1095)
	at org.jooq.impl.DefaultBinding$DefaultArrayBinding.get0(DefaultBinding.java:1032)
	at org.jooq.impl.DefaultBinding$DefaultArrayBinding.get0(DefaultBinding.java:911)
	at org.jooq.impl.DefaultBinding$AbstractBinding.get(DefaultBinding.java:774)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1720)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.operate(CursorImpl.java:1689)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.operate(CursorImpl.java:1654)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:125)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1618)
	at org.jooq.impl.CursorImpl$CursorIterator.hasNext(CursorImpl.java:1585)
	at org.jooq.impl.CursorImpl.fetchNext(CursorImpl.java:407)
	at org.jooq.impl.Tools.fetchOne(Tools.java:1762)
	at org.jooq.impl.AbstractResultQuery.fetchOne(AbstractResultQuery.java:545)
	at org.jooq.impl.AbstractResultQuery.fetchOne(AbstractResultQuery.java:550)
	at org.jooq.impl.SelectImpl.fetchOne(SelectImpl.java:2884)
Caused by: java.lang.NullPointerException: null
	at org.jooq.impl.Fields.size(Fields.java:81)
	at org.jooq.impl.RowImpl.size(RowImpl.java:224)
	at org.jooq.impl.AbstractRecord.<init>(AbstractRecord.java:136)
	at org.jooq.impl.RecordImpl.<init>(RecordImpl.java:131)
	at org.jooq.impl.Tools$1.newInstance(Tools.java:782)
	at org.jooq.impl.Tools.newRecord(Tools.java:756)
	... 88 common frames omitted
```

### Versions:

- jOOQ: 3.11.4
- Java: zulu-10
- Database (include vendor): postgres 9.6
- OS: mac osx (docker)

**Comment by lukaseder** [comment]
Thanks for your report. Probably a duplicate of https://github.com/jOOQ/jOOQ/issues/7691

**Comment by lukaseder** [comment]
Yes, can confirm. Your suggested fix is the one that was implemented in #7691. Thanks again for reporting!
